### ISSUE TITLE
option '-configuration' may only be provided once

### ISSUE THREAD

**Issue by suihan** [fastlane/fastlane#7186]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [ ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
my lane script for Inhouse is like this,and it works well when I was in fastlane[version 108],when I update to version 110.it breaks.
```
gym(scheme: "Mogujie4iPhone-WithoutWatch",use_legacy_build_api: true,output_directory:"./build",archive_path:"./build/Mogujie4iPhone.xcarchive",configuration:"CI")

```
$ set -o pipefail && xcodebuild -workspace ./Mogujie4iPhone.xcworkspace -scheme Mogujie4iPhone-WithoutWatch -configuration CI -configuration 'CI' -destination 'generic/platform=iOS' -archivePath ./build/Mogujie4iPhone.xcarchive archive | tee /Users/mogujie/Library/Logs/gym/Mogujie4iPhone-Mogujie4iPhone-WithoutWatch.log | xcpretty

### Environment
[19:39:04]: Generating fastlane environment output, this might take a few seconds...
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                              |
| ------------------- | -------------------------------------------------- |
| OS                  | 10.12                                              |
| Ruby                | 2.2.4                                              |
| Bundler?            | false                                              |
| Git                 | git version 2.9.3 (Apple Git-75)                   |
| Installation Source | /Users/mogujie/.rvm/rubies/ruby-2.2.4/bin/fastlane |
| Host                | Mac OS X 10.12 (16A323)                            |
| Ruby Lib Dir        | /Users/mogujie/.rvm/rubies/ruby-2.2.4/lib          |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                        |
| Is contained        | false                                              |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/        |
| Xcode Version       | 8.1                                                |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | zh_CN.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem                 | Version | Update-Status       |
| ------------------- | ------- | ------------------- |
| credentials_manager | 0.16.2  | ✅ Up-To-Date        |
| spaceship           | 0.37.0  | 🚫 Update availaible |
| fastlane_core       | 0.57.1  | ✅ Up-To-Date        |
| deliver             | 1.15.1  | ✅ Up-To-Date        |
| snapshot            | 1.16.3  | 🚫 Update availaible |
| frameit             | 3.0.0   | ✅ Up-To-Date        |
| pem                 | 1.4.0   | ✅ Up-To-Date        |
| cert                | 1.4.4   | ✅ Up-To-Date        |
| sigh                | 1.11.2  | ✅ Up-To-Date        |
| produce             | 1.3.0   | 🚫 Update availaible |
| gym                 | 1.12.0  | 🚫 Update availaible |
| pilot               | 1.12.1  | ✅ Up-To-Date        |
| scan                | 0.14.1  | 🚫 Update availaible |
| supply              | 0.7.1   | ✅ Up-To-Date        |
| match               | 0.11.0  | ✅ Up-To-Date        |
| screengrab          | 0.5.5   | 🚫 Update availaible |
| fastlane            | 1.110.0 | ✅ Up-To-Date        |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.13.6       |
| highline                  | 1.7.8        |
| commander                 | 4.4.0        |
| terminal-table            | 1.4.5        |
| json                      | 1.8.1        |
| multipart-post            | 2.0.0        |
| faraday                   | 0.9.2        |
| faraday_middleware        | 0.10.0       |
| dotenv                    | 2.1.1        |
| net-ssh                   | 3.2.0        |
| net-sftp                  | 2.1.2        |
| plist                     | 3.1.0        |
| rubyzip                   | 1.1.7        |
| security                  | 0.1.3        |
| krausefx-shenzhen         | 0.14.11      |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.3        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.2        |
| xcodeproj                 | 1.4.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.0        |
| credentials_manager       | 0.16.2       |
| multi_xml                 | 0.5.5        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161021 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.6.8        |
| xcode-install             | 2.0.8        |
| word_wrap                 | 1.0.0        |
| multi_json                | 1.12.1       |
| babosa                    | 1.0.2        |
| excon                     | 0.54.0       |
| gh_inspector              | 1.0.2        |
| fastlane_core             | 0.57.1       |
| mini_magick               | 4.5.1        |
| xcpretty-travis-formatter | 0.0.4        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.2.4      |
| google-api-client         | 0.9.20       |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.4.3        |
| activesupport             | 4.2.7.1      |
</details>

*generated on:* **2016-11-26**
</details>

**Comment by suihan** [comment]
Using legacy build system - waiting for radar to be fixed: https://openradar.appspot.com/radar?id=4952000420642816
[19:51:57]: $ xcodebuild -list -workspace ./Mogujie4iPhone.xcworkspace -configuration CI
[19:52:00]: $ xcodebuild clean -showBuildSettings -workspace ./Mogujie4iPhone.xcworkspace -scheme Mogujie4iPhone-WithoutWatch -configuration CI

+----------------------+----------------------------------+
|                 Summary for gym 1.12.0                  |
+----------------------+----------------------------------+
| scheme               | Mogujie4iPhone-WithoutWatch      |
| configuration        | CI                               |
| use_legacy_build_api | true                             |
| output_directory     | ./build                          |
| archive_path         | ./build/Mogujie4iPhone.xcarchive |
| workspace            | ./Mogujie4iPhone.xcworkspace     |
| destination          | generic/platform=iOS             |
| output_name          | Mogujie4iPhone                   |
| clean                | false                            |
| silent               | false                            |
| buildlog_path        | ~/Library/Logs/gym               |
| xcode_path           | /Applications/Xcode.app          |
+----------------------+----------------------------------+

[19:52:03]: $ set -o pipefail && xcodebuild -workspace ./Mogujie4iPhone.xcworkspace -scheme Mogujie4iPhone-WithoutWatch -configuration CI -configuration 'CI' -destination 'generic/platform=iOS' -archivePath ./build/Mogujie4iPhone.xcarchive archive | tee /Users/mogujie/Library/Logs/gym/Mogujie4iPhone-Mogujie4iPhone-WithoutWatch.log | xcpretty
[19:52:04]: ▸ xcodebuild: error: option '-configuration' may only be provided once
[19:52:04]: ▸ Usage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
[19:52:04]: ▸ xcodebuild [-project <projectname>] -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
[19:52:04]: ▸ xcodebuild -workspace <workspacename> -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
[19:52:04]: ▸ xcodebuild -version [-sdk [<sdkfullpath>|<sdkname>] [<infoitem>] ]
[19:52:04]: ▸ xcodebuild -list [[-project <projectname>]|[-workspace <workspacename>]] [-json]
[19:52:04]: ▸ xcodebuild -showsdks
[19:52:04]: ▸ xcodebuild -exportArchive -archivePath <xcarchivepath> -exportPath <destinationpath> -exportOptionsPlist <plistpath>
[19:52:04]: ▸ xcodebuild -exportLocalizations -localizationPath <path> -project <projectname> [-exportLanguage <targetlanguage>...]
[19:52:04]: ▸ xcodebuild -importLocalizations -localizationPath <path> -project <projectname>
[19:52:04]: ▸ Options:
[19:52:04]: ▸ -usage                              print brief usage
[19:52:04]: ▸ -help                               print complete usage
[19:52:04]: ▸ -verbose                            provide additional status output
[19:52:04]: ▸ -license                            show the Xcode and SDK license agreements
[19:52:04]: ▸ -checkFirstLaunchStatus             Check if any First Launch tasks need to be performed
[19:52:04]: ▸ -project NAME                       build the project NAME
[19:52:04]: ▸ -target NAME                        build the target NAME
[19:52:04]: ▸ -alltargets                         build all targets
[19:52:04]: ▸ -workspace NAME                     build the workspace NAME
[19:52:04]: ▸ -scheme NAME                        build the scheme NAME
[19:52:04]: ▸ -configuration NAME                 use the build configuration NAME for building each target
[19:52:04]: ▸ -xcconfig PATH                      apply the build settings defined in the file at PATH as overrides
[19:52:04]: ▸ -arch ARCH                          build each target for the architecture ARCH; this will override architectures defined in the project
[19:52:04]: ▸ -sdk SDK                            use SDK as the name or path of the base SDK when building the project
[19:52:04]: ▸ -toolchain NAME                     use the toolchain with identifier or name NAME
[19:52:04]: ▸ -destination DESTINATIONSPECIFIER   use the destination described by DESTINATIONSPECIFIER (a comma-separated set of key=value pairs describing the destination to use)
[19:52:04]: ▸ -destination-timeout TIMEOUT        wait for TIMEOUT seconds while searching for the destination device
[19:52:04]: ▸ -parallelizeTargets                 build independent targets in parallel
[19:52:04]: ▸ -jobs NUMBER                        specify the maximum number of concurrent build operations
[19:52:04]: ▸ -dry-run                            do everything except actually running the commands
[19:52:04]: ▸ -quiet                              do not print any output except for warnings and errors
[19:52:04]: ▸ -hideShellScriptEnvironment         don't show shell script environment variables in build log
[19:52:04]: ▸ -showsdks                           display a compact list of the installed SDKs
[19:52:04]: ▸ -showBuildSettings                  display a list of build settings and values
[19:52:04]: ▸ -list                               lists the targets and configurations in a project, or the schemes in a workspace
[19:52:04]: ▸ -find-executable NAME               display the full path to executable NAME in the provided SDK and toolchain
[19:52:04]: ▸ -find-library NAME                  display the full path to library NAME in the provided SDK and toolchain
[19:52:04]: ▸ -version                            display the version of Xcode; with -sdk will display info about one or all installed SDKs
[19:52:04]: ▸ -enableAddressSanitizer YES|NO      turn the address sanitizer on or off
[19:52:04]: ▸ -enableThreadSanitizer YES|NO       turn the thread sanitizer on or off
[19:52:04]: ▸ -resultBundlePath PATH              specifies the directory where a result bundle describing what occurred will be placed
[19:52:04]: ▸ -derivedDataPath PATH               specifies the directory where build products and other derived data will go
[19:52:04]: ▸ -archivePath PATH                   specifies the directory where any created archives will be placed, or the archive that should be exported
[19:52:04]: ▸ -exportArchive                      specifies that an archive should be exported
[19:52:04]: ▸ -exportOptionsPlist PATH            specifies a path to a plist file that configures archive exporting
[19:52:04]: ▸ -enableCodeCoverage YES|NO          turn code coverage on or off when testing
[19:52:04]: ▸ -exportPath PATH                    specifies the destination for the product exported from an archive
[19:52:04]: ▸ -skipUnavailableActions             specifies that scheme actions that cannot be performed should be skipped instead of causing a failure
[19:52:04]: ▸ -exportLocalizations                exports completed and outstanding project localizations
[19:52:04]: ▸ -importLocalizations                imports localizations for a project, assuming any necessary localized resources have been created in Xcode
[19:52:04]: ▸ -localizationPath                   specifies a path to XLIFF localization files
[19:52:04]: ▸ -exportLanguage                     specifies multiple optional ISO 639-1 languages included in a localization export
[19:52:04]: ▸ -xctestrun                          specifies a path to a test run specification
[19:52:04]: ▸ -only-testing:TEST-IDENTIFIER       constrains testing by specifying tests to include, and excluding other tests
[19:52:04]: ▸ -skip-testing:TEST-IDENTIFIER       constrains testing by specifying tests to exclude, but including other tests
[19:52:04]: ▸ -json                               output as JSON (note: -json implies -quiet)
xcodebuild: error: option '-configuration' may only be provided once

**Comment by suihan** [comment]
why the xcodebuild -list action add -configuration CI，I have two server，one add configuration automatic，another do not add it。without the configuration，it success

**Comment by hjanuschka** [comment]
@suihan please try to upgrade all fastlane gems, this issue should have been fixed with: https://github.com/fastlane/fastlane/pull/7108

and released with gym `1.12.1` https://github.com/fastlane/fastlane/releases/tag/gym%2F1.12.1

**Comment by suihan** [comment]
it works,thx
